about/about-kubernetes-egress.md

```diff
@@ -0,0 +1,111 @@
+- What is "NAT outgoing" and when is it used?
+- What is an egress gateway, and why might you want to use one?
+
+### What is Kubernetes egress?
+
+In this guide we are using the term Kubernetes egress to describe connections being made from pods to anything outside of the cluster.
+
+In contrast to ingress traffic, where Kubernetes has the [Ingress]({{site.baseurl}}/about/about-kubernetes-ingress)
+resource type to help manage the traffic, there is no Kubernetes Egress resource. Instead, how the egress traffic is
+handled at a networking level is determined by the Kubernetes network implementation / CNI plugin being used by the
+cluster. In addition, if a service mesh is being used, this can add additional egress behaviors on top of what the
+network implementation provides.
+
+There are three areas of behavior worth understanding for egress traffic, so you can choose a networking and/or service
+mesh setup that best suits your needs:
+- Restricting egress traffic
+- Outgoing NAT behavior
+- Egress gateways
+
+### Restricting egress traffic
+
+It's a common security requirement and best practice to restrict outgoing connections from the cluster. This is normally
+achieved using [Network Policy]({{site.baseurl}}/about/about-network-policy) to define egress rules for each
+microservices, often in conjunction with a [default deny]({{site.baseurl}}/about/about-network-policy#default-deny)
+policy that ensures outgoing connections are denied by default, until a policy is defined to explicitly allow specific
+traffic.
+
+One limitation when using Kubernetes Network Policy to restrict access to specific external resources, that the external
+resources need to be specified as IP addresses (or IP address ranges) within the policy rules. If the IP addresses
+associated with an external resource change, then every policy that referenced those IP addresses needs to updated with
+the new IP addresses. This limitation can be circumvented using Calico [Network
+Sets]({{site.baseurl}}/security/external-ips-policy), or Calico Enterprise's support for [domain
+names]({{site.baseurl}}/security/calico-enterprise/egress-access-controls) in policy rules.
+
+In addition to using network policy, service meshes typically allow you to configure which external services each pod
+can access. In the case of Istio, {{site.prodname}} can be integrated to enforce network policy at the service mesh
+layer, including [L5-7 rules]({{site.baseurl}}/security/http-methods), as another alternative to using IP addresses in rules. To
+learn more about the benefits of this kind of approach, read our [Adopt a zero trust network model for security
+]({{site.baseurl}}/security/adopt-zero-trust) guide.
+
+Note in addition to everything mentioned so far, perimeter firewalls can also be used to restrict outgoing connections,
+for example to only allow connections to particular external IP address ranges, or external services. However, since
```

Comment:
for example to allow connections **only** to particular external IP address ranges,